src/shared/utils/roles.ts

```diff
@@ -44,7 +44,10 @@ export async function setN0llanRole(user: User, kthId: string): Promise<void> {
 	}
 }
 
-export async function setYearRole(user: User, kthId: string): Promise<void> {
+export async function setYearRole(
+	user: User,
+	kthId: string
+): Promise<number | null> {
```

Comment:
As we said, do not return the number here. Break out the logic and remove this function